### PR TITLE
feat: add extractor for ebay video listings

### LINKED ISSUE #6134: Support for eBay listing videos
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2023.01.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

_No response_

### Example URLs

https://www.ebay.com/itm/194509326719

### Provide a description that is worded well enough to be understood

eBay apparently have begun supporting videos in the listing gallery, mixed in with the photos. Someone sent me a URL (provided in the Example URLs) which contains one such video. It's the second item in the gallery (click the thumbnail with the play button on the left of the main image).

Tested with yt-dlp (latest) but appears there's no support yet. The browser network log shows M4S segmented video/audio streams being downloaded when played.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-F', 'https://www.ebay.com/itm/194509326719']
[debug] Portable config "D:\Global\yt-dlp.conf": ['-o', 'M:\\Videos\\%(title)s [%(uploader)s, %(extractor)s].%(ext)s', '--console-
title', '--no-mtime', '--add-metadata', '--compat-options', 'all', '--no-clean-infojson', '--compat-option', 'filename-sanitizatio
n']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2023.01.06 [6becd25] (win_exe)
[debug] Compatibility options: no-external-downloader-progress, format-spec, list-formats, multistreams, no-live-chat, no-direct-m
erge, embed-metadata, embed-thumbnail-atomicparsley, *filename, no-youtube-channel-redirect, filename-sanitization, playlist-index
, no-certifi, no-playlist-metafiles, format-sort, *no-clean-infojson, abort-on-error, no-youtube-prefer-utc-upload-date, seperate-
video-versions, no-attach-info-json, no-keep-subs, no-youtube-unavailable-videos
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-8.1-6.3.9600-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-105349-g0a83ecbf48-20220124 (setts), ffprobe N-105349-g0a83ecbf48-20220124, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1760 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.01.06, Current version: 2023.01.06
yt-dlp is up to date (2023.01.06)
[generic] Extracting URL: https://www.ebay.com/itm/194509326719
[generic] 194509326719: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 194509326719: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.ebay.com/itm/194509326719
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1502, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1578, in __extract_info
  File "yt_dlp\extractor\common.py", line 680, in extract
  File "yt_dlp\extractor\generic.py", line 2523, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.ebay.com/itm/194509326719
```

### PR DESCRIPTION
**Pull request by JChris246**
### Description of your *pull request* and other information

This PR seeks to add an extractor for ebay video listings as described in issue #6134

Fixes #6134 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/ebay.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            # m3u8 download
+            'skip_download': True,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        video_json = self._search_json(r'"video":\s*', webpage, "video_json", video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/ebay.py:24]:
```suggestion
        video_json = self._search_json(r'"video":', webpage, 'video json', video_id)
```

File: yt_dlp/extractor/ebay.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            raise ExtractorError('Unable to extract video urls')
+
+        formats = []
+        for key in playlist_map:
+            if key == "HLS":
+                formats.extend(self._extract_m3u8_formats(
+                    playlist_map.get(key), video_id, m3u8_id='hls', fatal=False))
+            if key == "DASH":
+                formats.extend(self._extract_mpd_formats(
+                    playlist_map.get(key), video_id, fatal=False))
```
Comment (pukkandan) [inline yt_dlp/extractor/ebay.py:37]:
```suggestion
        formats = []
        for key, url in video_json['playlistMap'].items():
            if key == 'HLS':
                formats.extend(self._extract_m3u8_formats(url, video_id, fatal=False))
            elif key == 'DASH':
                formats.extend(self._extract_mpd_formats(url, video_id, fatal=False))
            else:
                # Add a warning here
```

File: yt_dlp/extractor/ebay.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    _TEST = {
+        'url': 'https://www.ebay.com/itm/194509326719',
+        'info_dict': {
+            'id': '194509326719',
+            'ext': 'mp4',
+            'title': 'HELIX upload [AN_Master_A04B.mp4] - 1666069406072',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
```
Comment (pukkandan) [inline yt_dlp/extractor/ebay.py:16]:
```suggestion
            'skip_download': 'm3u8'',
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/ebay.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        'info_dict': {
+            'id': '194509326719',
+            'ext': 'mp4',
+            'title': 'HELIX upload [AN_Master_A04B.mp4] - 1666069406072',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': 'm3u8',
+        }
+    }
```
Comment (pukkandan) [inline yt_dlp/extractor/ebay.py:18]:
```suggestion
    _TESTS = [{
        'url': 'https://www.ebay.com/itm/194509326719',
        'info_dict': {
            'id': '194509326719',
            'ext': 'mp4',
            'title': 'HELIX upload [AN_Master_A04B.mp4] - 1666069406072',
        },
        'params': {'skip_download': 'm3u8'}
    }]
```

File: yt_dlp/extractor/ebay.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/ebay.py]: quotes unresolved

File: yt_dlp/extractor/ebay.py (hunk 2 above)
Comment (pukkandan) [reply yt_dlp/extractor/ebay.py]: You only applied half of my suggestion. `if not playlist_map` isn't needed

File: yt_dlp/extractor/ebay.py (hunk 3 above)
Comment (pukkandan) [reply yt_dlp/extractor/ebay.py]: again, only applying half!

**Review by pukkandan — CHANGES_REQUESTED** [review]
Does a product only ever have a single video? If so, isn't it better to use the webpage's title instead of `video_json.title`?

**Comment by JChris246** [comment]
> Does a product only ever have a single video? If so, isn't it better to use the webpage's title instead of `video_json.title`?

I wasn't able to find any where there was more than one video, so I supposed it's safe to assume it they all only have a single video. I made the change to use the html title instead 👍

File: yt_dlp/extractor/ebay.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            if key == 'HLS':
+                formats.extend(self._extract_m3u8_formats(url, video_id, fatal=False))
+            elif key == 'DASH':
+                formats.extend(self._extract_mpd_formats(url, video_id, fatal=False))
+            else:
+                self.report_warning('unsupported format found: %s' % key, video_id)
+
+        return {
+            'id': video_id,
+            'title': self._html_extract_title(webpage),
```
Comment (pukkandan) [inline yt_dlp/extractor/ebay.py:33]:
```suggestion
            'title': remove_end(self._html_extract_title(webpage), ' | eBay'),
```

File: yt_dlp/extractor/ebay.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        video_json = self._search_json(r'"video":', webpage, 'video json', video_id)
+
+        formats = []
+        for key, url in video_json['playlistMap'].items():
+            if key == 'HLS':
+                formats.extend(self._extract_m3u8_formats(url, video_id, fatal=False))
+            elif key == 'DASH':
+                formats.extend(self._extract_mpd_formats(url, video_id, fatal=False))
+            else:
+                self.report_warning('unsupported format found: %s' % key, video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/ebay.py:29]:
```suggestion
                self.report_warning(f'Unsupported format {key}', video_id)
```

**Review by pukkandan — APPROVED** [review]

**Pull request merged**